### PR TITLE
Fix exclusive option when dataset is not set

### LINKED ISSUE #514: [BUG]NPE in Alluxio Runtime Controller
**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**

```
E1229 13:56:23.610118       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 263 [running]:
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x14b87a0, 0x24d5fc0)
        /go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x14b87a0, 0x24d5fc0)
        /usr/local/go/src/runtime/panic.go:969 +0x166
github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio.Build(0xc0004fcbb0, 0xb, 0x18a04a0, 0xc00004a018, 0xc0004fc9f9, 0x7, 0xc0004fc9ec, 0x3, 0x18a7200, 0xc00061c380, ...)
        /go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio/engine.go:59 +0x118
github.com/fluid-cloudnative/fluid/pkg/ddc.CreateEngine(0xc0004fcbb0, 0xb, 0x18a04a0, 0xc00004a018, 0xc0004fc9f9, 0x7, 0xc0004fc9ec, 0x3, 0x18a7200, 0xc00061c380, ...)
        /go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/factory.go:42 +0xc3
github.com/fluid-cloudnative/fluid/pkg/controllers/v1alpha1/alluxio.(*RuntimeReconciler).GetOrCreateEngine(0xc00030f8e0, 0x18a04a0, 0xc00004a018, 0xc0004fc9f9, 0x7, 0xc0004fc9ec, 0x3, 0x18a7200, 0xc00061c380, 0x0, ...)
```

**What you expect to happen:**

**How to reproduce it**

**Additional Information**

### PR DESCRIPTION
**Pull request by cheyang**
### Ⅱ. Does this pull request fix one issue?

fixes #514

### Ⅴ. Special notes for reviews

### CODE REVIEW

**Review by xieydd — APPROVED** [review]
/lgtm, how about fix the ci error

**Comment by cheyang** [comment]
> /lgtm, how about fix the ci error

Done

File: pkg/ddc/alluxio/engine.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ type AlluxioEngine struct {
 	initImage              string
 	MetadataSyncDoneCh     chan MetadataSyncResult
 	runtimeInfo            base.RuntimeInfoInterface
-	exclusive              bool
+	UnitTest               bool
```
Comment (xieydd) [inline pkg/ddc/alluxio/engine.go:45]: a little confused? why add this

Comment (cheyang) [reply pkg/ddc/alluxio/engine.go:45]: It's for testing.

Comment (cheyang) [reply pkg/ddc/alluxio/engine.go:45]:
```go
if !e.UnitTest {
		dataset, err := utils.GetDataset(e.Client, e.name, e.namespace)
		if err != nil {
			e.Log.Info("Failed to get dataset when getruntimeInfo")
			return e.runtimeInfo, err
		}

		e.runtimeInfo.SetupWithDataset(dataset)
	}

```

**Pull request merged**